src/Grpc.AspNetCore.Server/GrpcServiceOptions.cs

```diff
@@ -73,6 +76,34 @@ public List<ICompressionProvider> CompressionProviders
             }
             set => _compressionProviders = value;
         }
+
+        /// <summary>
+        /// List of interceptors to be executed. Interceptors executed in order.
+        /// </summary>
+        internal List<Type> Interceptors { get; private set; } = new List<Type>();
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        public void AddInterceptor<T>()
+        {
+            AddInterceptor(typeof(T));
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="interceptors"></param>
+        public void AddInterceptor(params Type[] interceptors)
```

Comment:
IMO remove the params and just accept one interceptor. Basically be a non-generic version of `AddInterceptor<T>()`